### PR TITLE
Update commons-ui to v0.51 

### PR DESCRIPTION
**Pull request by flomillot**
Also use the real types from the lib and clean associated code

### CODE REVIEW

File: src/routes/router.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ export function appRoutes(): RouteObject[] {
 }
 
 const AuthRouter: FunctionComponent<{
-    userManager: (typeof AuthenticationRouter)['userManager'];
+    userManager: Parameters<typeof AuthenticationRouter>[0]['userManager'];
```
Comment (Tristan-WorkGH) [inline src/routes/router.tsx:82]: I see with @TheMaskedTurtle, who find the previous version more readable (I did your version before).

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
         "ag-grid-react": "^31.1.1",
         "core-js": "^3.6.4",
         "notistack": "^3.0.0",
+        "oidc-client": "^1.10.1",
```
Comment (Tristan-WorkGH) [inline package.json:35]: Isn't it a commons-ui dependency?

File: src/routes/router.tsx (hunk 1 above)
Comment (flomillot) [reply src/routes/router.tsx:82]: We tried it does not work

File: package.json (hunk 2 above)
Comment (flomillot) [reply package.json:35]: But we import it directly in the the project so we need to add it in the deps

File: src/routes/router.tsx (hunk 1 above)
Comment (flomillot) [reply src/routes/router.tsx:82]: (plus I find this version more understandable)

Comment (Tristan-WorkGH) [reply src/routes/router.tsx:82]: okay

**Review by Tristan-WorkGH — APPROVED** [review]

**Pull request merged**